docs/source/manual/configuration.rst

```diff
@@ -1313,6 +1314,7 @@ Name                         Default            Description
 protocol                     TLSv1.2            The default protocol the client will attempt to use during the SSL Handshake.
                                                 See
                                                 `here <http://docs.oracle.com/javase/8/docs/technotes/guides/security/StandardNames.html#SSLContext>`_ for more information.
+provider                     (none)             The name of the JCE provider to use on client side for cryptographic support (for example, SunJCE, Conscrypt, BC, etc).
```

Comment:
Are these really valid values for the `SSLContextBuilder#setProvider()` method?

And if so, why is "SUN" being used in the example above?

Reply:
`SSLContextBuilder#setProvider()` calls `Security.getProvider()` under the hood.
```java
    public SSLContextBuilder setProvider(final String name) {
        this.provider = Security.getProvider(name);
        return this;
    }
```
Oracle Standard JDK comes with 10 providers, SUN and SunJCE among them. SUN is what `Security.getProviders()[].getName()` returns on my JDK. Index 0 means it has the highest priority. That's why "SUN" in the example above. But I think now that it's better to specify default value as `SunJSSE`, as the provider which will be used in fact with TLSv1.2 (in JDK 8) is SunJSSE.
`BC` and `Conscrypt` will work if they are previously configured using `Security.addProvider()` or `Security.insertProviderAt`.

Reply:
Thanks for the explanation! 👍

Please add a link to https://docs.oracle.com/javase/8/docs/technotes/guides/security/SunProviders.html for a more detailed overview.